From 3fcd5f431dee9909b6bcbd9b8b61d4b1fe4b5f92 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Fri, 12 Nov 2021 18:01:05 +0100
Subject: [PATCH] meson.build: add girdir to gio-2.0.pc and glib-2.0.pc

Add girdir to gio-2.0.pc and glib-2.0.pc to fix the following build
failure with atk or libnice:

/home/giuliobenetti/autobuild/run/instance-1/output-1/host/riscv32-buildroot-linux-gnu/sysroot/usr/bin/g-ir-compiler atk/Atk-1.0.gir --output atk/Atk-1.0.typelib --includedir=/usr/share/gir-1.0
Could not find GIR file 'GObject-2.0.gir'; check XDG_DATA_DIRS or use --includedir

Fixes:
 - http://autobuild.buildroot.org/results/2716929db638977e6bf665352a08bd580e1dd0ae
 - http://autobuild.buildroot.org/results/3088ef32b03e0bb984291b1227b187f1ff816eb7

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Dario: make the patch to be applied with fuzz factor 0]
Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>
[Fiona: refresh for glib 2.80.4]
Signed-off-by: Fiona Klute (WIWA) <fiona.klute@gmx.de>
Upstream: Never submitted
---
 gio/meson.build  | 1 +
 glib/meson.build | 1 +
 2 files changed, 2 insertions(+)

diff --git a/gio/meson.build b/gio/meson.build
index 59c2b0f..5120537 100644
--- a/gio/meson.build
+++ b/gio/meson.build
@@ -885,6 +885,7 @@ pkg.generate(libgio,
   variables : [
     'schemasdir=' + '${datadir}' / schemas_subdir,
     'dtdsdir=' + '${datadir}' / dtds_subdir,
+    'girdir=' + join_paths('${libdir}', '../share/gir-1.0'),
     'giomoduledir=' + pkgconfig_giomodulesdir,
     'gio=' + '${bindir}' / 'gio',
     'gio_querymodules=' + pkgconfig_multiarch_bindir / 'gio-querymodules',
diff --git a/glib/meson.build b/glib/meson.build
index d2efeba..a69532b 100644
--- a/glib/meson.build
+++ b/glib/meson.build
@@ -447,6 +447,7 @@ pkg.generate(libglib,
   subdirs : ['glib-2.0'],
   extra_cflags : ['-I${libdir}/glib-2.0/include'] + win32_cflags,
   variables : [
+    'girdir=' + join_paths('${libdir}', '../share/gir-1.0'),
     'glib_genmarshal=' + '${bindir}' / 'glib-genmarshal',
     'gobject_query=' + '${bindir}' / 'gobject-query',
     'glib_mkenums=' + '${bindir}' / 'glib-mkenums',
-- 
2.45.2

